xonsh/aliases.py

```diff
@@ -54,6 +55,40 @@ def EXEC_ALIAS_RE():
     return re.compile(r"@\(|\$\(|!\(|\$\[|!\[|\&\&|\|\||\s+and\s+|\s+or\s+|[>|<]")
 
 
+class FuncAlias:
+    """Provides a callable alias for xonsh commands."""
+
+    attributes_inherit = ["__xonsh_threadable__", "__xonsh_capturable__", "__doc__"]
+    attributes_show = ["__xonsh_threadable__", "__xonsh_capturable__"]
+
+    def __init__(self, name, func):
+        self.__name__ = self.name = name
+        self.func = func
+        for attr in self.attributes_inherit:
+            if (val := getattr(func, attr, None)) is not None:
+                self.__setattr__(attr, val)
+
+    def __repr__(self):
+        r = {"name": self.name, "func": self.func}
+        r |= {
+            attr: val
+            for attr in self.attributes_show
+            if (val := getattr(self, attr, None)) is not None
+        }
+        return f"FuncAlias({repr(r)})"
+
+    def __call__(
+        self, args=None, stdin=None, stdout=None, stderr=None, spec=None, stack=None
+    ):
+        try:
+            func_args = [args, stdin, stdout, stderr, spec, stack][
+                : len(inspect.signature(self.func).parameters)
+            ]
```

Comment:
Now that all the callable aliases would be called through `FuncAlias`, can we remove the Proxy handling for each number of arguments 1 to 5 ?